applications/crossbar/src/modules/cb_transactions.erl

```diff
@@ -101,8 +113,66 @@ validate(Context) ->
 validate(Context, PathToken) ->
     validate_transaction(Context, PathToken, cb_context:req_verb(Context)).
 
--spec validate_transactions(cb_context:context(), http_method()) ->
-                                   cb_context:context().
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec delete(cb_context:context(), path_token()) -> cb_context:context().
+delete(Context, ?DEBIT) ->
+    case cb_context:resp_status(Context) of
+        'success' ->
+            case create_debit_tansaction(Context) of
```

Comment:
Let's put this clause in a function, perhaps called `maybe_create_debit_transaction(Context)`